xarray/tests/test_plot.py

```diff
@@ -2381,3 +2382,50 @@ def test_facetgrid_single_contour():
     ds["time"] = [0, 1]
 
     ds.plot.contour(col="time", levels=[4], colors=["k"])
+
+
+@requires_matplotlib
+@pytest.mark.parametrize("figsize", [[4, 4], None])
+@pytest.mark.parametrize("aspect", [4 / 3, None])
+@pytest.mark.parametrize("size", [200, None])
+def test_get_axis(figsize, aspect, size):
+    # test get_axis works with different args combinations
+    # and return the right type
+
+    ax = "something"
+    if figsize is not None:
+        # cannot provide both ax and figsize
+        with pytest.raises(ValueError, match="both `figsize` and `ax`"):
+            ax = get_axis(figsize, size, aspect, ax)
+    if figsize is None and size is not None:
+        # cannot provide both ax and size
+        with pytest.raises(ValueError, match="both `size` and `ax`"):
+            ax = get_axis(figsize, size, aspect, ax)
+
+    ax = None
+    if figsize is not None and size is not None:
+        # cannot provide both size and figsize
+        with pytest.raises(ValueError, match="both `figsize` and " "`size`"):
+            ax = get_axis(figsize, size, aspect, ax)
+
+    if figsize is None and aspect is not None and size is None:
+        # cannot provide aspect and size
+        with pytest.raises(ValueError, match="`aspect` argument without `size`"):
+            ax = get_axis(figsize, size, aspect, ax)
+
+    ax = plt.axes()
+    axtype = type(ax)
+    ax = get_axis(ax=ax)
+    assert isinstance(ax, axtype)
+    ax = get_axis()
+    assert isinstance(ax, axtype)
+
+    try:
+        import cartopy as ctpy  # type: ignore
```

Comment:
Let's move the cartopy bit to its own test. It's now running 8 times though it isn't using any of the parameters in the `parametrize` statements :)

I think you can use `ctpy = pytest.importorskip("cartopy")` to skip the test if `cartopy` is not installed.

Reply:
either that or add a line here: https://github.com/pydata/xarray/blob/7ae74feb9a96e0ad0ca12f33fdb3b625ff58f924/xarray/tests/__init__.py#L57 and then use `requires_cartopy` as a decorator. I actually would prefer that since then reading the test body is not required to know why the test got skipped.

Reply:
done